modules/imgproc/perf/perf_integral.cpp

```diff
@@ -31,28 +44,57 @@ PERF_TEST_P(Size_MatType_OutMatDepth, integral,
 }
 
 PERF_TEST_P(Size_MatType_OutMatDepth, integral_sqsum,
+                testing::Combine(
+                    testing::Values(TYPICAL_MAT_SIZES),
+                    testing::Values(CV_8UC1, CV_8UC2, CV_8UC3, CV_8UC4),
+                    testing::Values(CV_32S, CV_32F, CV_64F)
+                    )
+        )
+{
+    Size sz = get<0>(GetParam());
+    int matType = get<1>(GetParam());
+    int sdepth = get<2>(GetParam());
+
+    Mat src(sz, matType);
+    Mat sum(sz, sdepth);
+    Mat sqsum(sz, sdepth);
+
+    declare.in(src, WARMUP_RNG).out(sum, sqsum);
+    declare.time(100);
+
+    TEST_CYCLE() integral(src, sum, sqsum, sdepth);
+
+    SANITY_CHECK(sum, 1e-6);
+    SANITY_CHECK(sqsum, 1e-6);
+}
+
+int vals[6][2] = {{CV_32S, CV_32S}, {CV_32S, CV_32F}, {CV_32S, CV_64F}, {CV_32F, CV_32F}, {CV_32F, CV_64F}, {CV_64F, CV_64F}};
+PERF_TEST_P(Size_MatType_OutMatDepthArray, DISABLED_integral_sqsum_full,
             testing::Combine(
                 testing::Values(TYPICAL_MAT_SIZES),
                 testing::Values(CV_8UC1, CV_8UC2, CV_8UC3, CV_8UC4),
-                testing::Values(CV_32S, CV_32F, CV_64F)
+                testing::Values(0,1,2,3,4,5)
                 )
             )
 {
     Size sz = get<0>(GetParam());
     int matType = get<1>(GetParam());
-    int sdepth = get<2>(GetParam());
+    int *sizes = (int *)vals[get<2>(GetParam())];
+    int sdepth = sizes[0];
+    int sqdepth = sizes[1];
 
     Mat src(sz, matType);
     Mat sum(sz, sdepth);
-    Mat sqsum(sz, sdepth);
+    Mat sqsum(sz, sqdepth);
 
     declare.in(src, WARMUP_RNG).out(sum, sqsum);
     declare.time(100);
 
-    TEST_CYCLE() integral(src, sum, sqsum, sdepth);
+    TEST_CYCLE() integral(src, sum, sqsum, sdepth, sqdepth);
 
     SANITY_CHECK(sum, 1e-6);
     SANITY_CHECK(sqsum, 1e-6);
```

Comment:
You don't have sanity data for this test (in opencv_extra repositry). Replace these 2 lines by `SANITY_CHECK_NOTHING();`

Reply:
Thanks for the feedback, just made the changes you requested.